### PR TITLE
[api] Move `UGraph` out of API.

### PR DESCRIPTION
**Pull request by ejgallego**
We need to determine what part of the API should do in internal.
I'm afraid I am not qualified for that, so comments welcome.

c.f. issue #6008

### CODE REVIEW

File: kernel/uGraph.mli [hunk 1]
```diff
@@ -9,63 +9,72 @@
...
-val check_constraint  : universes -> univ_constraint -> bool
-val check_constraints : constraints -> universes -> bool
+include module type of Public
 
-val check_eq_instances : Instance.t check_function
-(** Check equality of instances w.r.t. a universe graph *)
+module Internal : sig
+
+(** The empty graph of universes *)
+val empty_universes : t
```
Comment (SkySkimmer) [inline kernel/uGraph.mli]: This is just an alias of initial_universes, I propose removing it.

File: kernel/uGraph.mli [hunk 2]
```diff
@@ -9,63 +9,72 @@
...
+
+val is_initial_universes : t -> bool
```
Comment (SkySkimmer) [inline kernel/uGraph.mli]: Why is this internal?

Comment (ejgallego) [reply kernel/uGraph.mli]: No reason, maybe it was not exported in API? Bear with me, the universe API is the thing I'm less familiar with the whole codebase. Making this public.

**Comment by ejgallego** [comment]
Both comments taken care of, thanks @SkySkimmer

**Comment by Zimmi48** [comment]
Finally, this PR contains a commit that is a good example of your approach (498eaed). But as I'm not familiar with the universes either, I can't comment much.

**Comment by ejgallego** [comment]
Ok, what should we do here? Something to keep internal?

File: kernel/uGraph.mli [hunk 3]
```diff
@@ -57,6 +57,8 @@ val empty_universes : t
 
 val sort_universes : t -> t
 
+module Internal : sig
+
 val constraints_of_universes : t -> Constraint.t
 
 val check_subtype : AUContext.t check_function
```
Comment (SkySkimmer) [inline kernel/uGraph.mli]: Not sure check_subtype should be internal but it should be fine.

**Review by SkySkimmer — APPROVED** [review]

**Comment by ejgallego** [comment]
Closing #6075 #6076 #6077 as it seems they are not gaining a lot of traction. In the past they used to be largish PRs, however, after the API removal all the PRs on the kernel are just a few lines, as these base libraries tend to have a full public interface. #6008 still applies thou.

**Pull request closed without merging**